pkg/detectors/abstract/abstract.go

```diff
@@ -55,7 +56,10 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			req.Header.Add("Content-Type", "application/json")
 			res, err := client.Do(req)
 			if err == nil {
-				defer res.Body.Close()
+				// Ensure we drain the response body so this connection can be reused.
+				_, _ = io.Copy(io.Discard, res.Body)
+				_ = res.Body.Close()
```

Comment:
If the body isn't being used, is there a reason to use `defer res.Body.Close()` (in an ugly anonymous function) instead of calling `res.Body.Close()` directly?